pkg/operation/botanist/controlplane.go

```diff
@@ -206,3 +212,27 @@ func (b *Botanist) RestartControlPlanePods(ctx context.Context) error {
 		client.MatchingLabels{v1beta1constants.LabelPodMaintenanceRestart: "true"},
 	)
 }
+
+func waitUntilNoPodsExistAnymore(ctx context.Context, c client.Client, namespace string, deployments []string) error {
+	fns := make([]flow.TaskFn, 0, len(deployments))
+	for _, deploymentName := range deployments {
+		fns = append(fns, func(ctx context.Context) error {
+			deployment := &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      deploymentName,
```

Comment:
The creation of the deployment object should be outside of the closure. The way that it is implemented now, if you have 3 deployments, all 3 executions will be done with the 3rd deployment name.

Reply:
Nice catch! Thanks